### DISCUSSION TITLE
Offscreen rendering for Hololens 2

### DISCUSSION THREAD

**Discussion by JPhilC** [cefsharp/CefSharp#3871] [Ideas]
I found CEF looking for a .NET wrapper for Chromium that would allow bitmaps to be generated offscreen. Specifically, I want to do this in an app running on Hololens 2.

For this to work I need to build for UWP on x64 (for testing on desktop) and Arm64 for running on the actual device.

I am happy to have a go at compiling the necessary sources targeting the platforms I need if someone could give me some pointers as to which projects I need to build or point me in the direction of some documentation.

Thanks

Phil

**Comment by amaitland** [comment]
I suspect you'll have to wait for Microsoft to provide a solution. 

See https://magpcss.org/ceforum/viewtopic.php?f=6&t=18634&p=50071

**Reply by amaitland** [reply]
You could potentially run chromium in headless mode on a server and stream the frames to your device. 

https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-startScreencast

Performance would obviously be a consideration.

**Reply by JPhilC** [reply]
Thanks for the reply, so it looks like CEF isn't the route to go, but possibly WebView2, according to the comments on the magpcss forum.  I'll look at WebView2.
